src/screens/TopicPageScreen/index.js

```diff
@@ -74,14 +71,54 @@ const TopicPageScreen = (props) => {
   const [offset, setOffset] = React.useState(0);
   const [client] = React.useContext(Context).client;
   const refBlockComponent = React.useRef();
-  const interactionManagerRef = React.useRef(null);
-  const interactionManagerAnimatedRef = React.useRef(null);
+  const scrollY = React.useRef(new Animated.Value(0)).current;
   const {mappingColorFeed} = useCoreFeed();
   const {listRef} = useOnBottomNavigationTabPressHook(LIST_VIEW_TYPE.TIKTOK_SCROLL, onRefresh);
 
   const topicWithPrefix = route.params.id;
   const id = removePrefixTopic(topicWithPrefix);
 
+  const coverPath = topicDetail?.cover_path || null;
+  const {top} = useSafeAreaInsets();
+  const topPosition = Platform.OS === 'ios' ? top : 0;
+
+  const navigationHeight = coverPath
+    ? dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT_COVER
+    : dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT;
+  const headerShowHeight =
+    navigationHeight + dimen.size.TOPIC_FEED_HEADER_HEIGHT + topPosition + normalize(4);
+  const headerHideHeight = dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT2 + topPosition;
+
+  const headerHeight = scrollY.interpolate({
+    inputRange: [0, headerHideHeight],
+    outputRange: [headerShowHeight, headerHideHeight],
+    extrapolate: 'clamp'
+  });
+
+  const opacityHeader = scrollY.interpolate({
+    inputRange: [0, 100],
+    outputRange: [1, 0],
+    extrapolate: 'clamp'
+  });
+
+  React.useEffect(() => {
+    const listener = scrollY.addListener(({value}) => {
+      if (value >= 0 && value <= headerHideHeight) {
+        setIsHeaderHide(false);
+      } else if (value >= headerHideHeight) {
+        setIsHeaderHide(true);
+      }
+    });
+
+    return () => {
+      scrollY.removeListener(listener);
+    };
+  }, [scrollY]);
+
+  const handleScroll = Animated.event([{nativeEvent: {contentOffset: {y: scrollY}}}], {
+    useNativeDriver: false
+  });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new implementation of the scroll event handling and header animation logic seems to be more efficient and cleaner than the previous one. However, ensure that it provides the same or better user experience.
```diff
-  const interactionManagerRef = React.useRef(null);
-  const interactionManagerAnimatedRef = React.useRef(null);
+  const scrollY = React.useRef(new Animated.Value(0)).current;
...
-  const handleOnScrollBeginDrag = (event) => {
-    lastDragY = event.nativeEvent.contentOffset.y;
-  };
-  const handleScrollEvent = React.useCallback(
-    (event) => {
-      const {y} = event.nativeEvent.contentOffset;
-      const dy = y - lastDragY;
-      if (y <= 30) {
-        showHeaderAnimation();
-      } else if (
-        dimen.size.TOPIC_CURRENT_ITEM_HEIGHT &&
-        dy - 20 > dimen.size.TOPIC_CURRENT_ITEM_HEIGHT / 2
-      ) {
-        interactionManagerAnimatedRef.current = InteractionManager.runAfterInteractions(() => {
-          Animated.timing(animatedHeight, {
-            toValue: dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT2 + topPosition,
-            duration: 100,
-            useNativeDriver: false
-          }).start();
-          Animated.timing(opacityHeaderAnimation, {
-            toValue: 0,
-            duration: 100,
-            useNativeDriver: false
-          }).start();
-        });
-        setIsHeaderHide(true);
-      }
-    },
-    [animatedHeight, coverPath]
-  );
+  const handleScroll = Animated.event([{nativeEvent: {contentOffset: {y: scrollY}}}], {
+    useNativeDriver: false
+  });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->